From: Jeremy Bicha <jeremy.bicha@canonical.com>
Date: Wed, 17 Jan 2024 13:07:26 -0500
Subject: driver: Allow closing the window in existing user mode

Existing user mode was disabled in 43f1870c
but some distros like Debian have reverted that change.

With existing user mode, Initial Setup is just another app.
Since the questions are not mandatory, it is reasonable
to not go out of our way to prevent users from exiting early.

This partially reverts 1db957a7

Forwarded: https://gitlab.gnome.org/GNOME/gnome-initial-setup/-/merge_requests/224
---
 gnome-initial-setup/gis-driver.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/gnome-initial-setup/gis-driver.c b/gnome-initial-setup/gis-driver.c
index 0b3f542..36b045b 100644
--- a/gnome-initial-setup/gis-driver.c
+++ b/gnome-initial-setup/gis-driver.c
@@ -802,10 +802,13 @@ gis_driver_startup (GApplication *app)
                     G_CALLBACK (window_realize_cb),
                     (gpointer)app);
 
-  g_signal_connect (driver->main_window,
-                    "close-request",
-                    G_CALLBACK (window_close_request_cb),
-                    NULL);
+  /* Only allow closing the window in existing user mode*/
+  if (driver->mode != GIS_DRIVER_MODE_EXISTING_USER) {
+    g_signal_connect (driver->main_window,
+                      "close-request",
+                      G_CALLBACK (window_close_request_cb),
+                      NULL);
+  }
 
   driver->assistant = g_object_new (GIS_TYPE_ASSISTANT, NULL);
   gtk_window_set_child (GTK_WINDOW (driver->main_window),
